### PR TITLE
Update for new Makie

### PR DESCRIPTION
**Pull request by asinghvi17**
Adds updates for Makie and the integration with MakieLayout.  Also updates examples, README, etc.  WIP.
download-1

cc @mkborregaard

### CODE REVIEW

**Comment by asinghvi17** [comment]
`with_theme(ggthemr(:fresh) do; MakieThemes.demofigure(); end`
download-4

New BBC plot theme inspired by bbplot:
download-3
download-2

**Comment by asinghvi17** [comment]
Figured out AoG and got the readme example in the same style as R!
download-4

**Comment by mkborregaard** [comment]
Oh wow amazing @asinghvi17 . Looks good, I'll check it more carefully when I get a change.
I actually had left this package a little by the wayside - please feel free to take over the main maintainership if you would like to.

**Comment by asinghvi17** [comment]
I was thinking of moving this to MakieOrg so it gets a little more visibility if that's alright with you @mkborregaard - I currently have use for a nice theme package for Makie, and will probably add modules for e.g. Business Standard or Bloomberg Quint.

**Comment by mkborregaard** [comment]
@asinghvi17 that's completely fine for me.
The only really "new" idea in this package that I'd like you to still consider is _the separation of line themes and color themes_. So many theme packages have color themes and what I call "line" themes mixed into one. "Line" themes are the fonts and font size, the appearance of grid lines, thickness and axes and whether they have arrows on them - in short, all of the "visual identity" of a given plotting package. By having themes for both colors and "lines", separately, Makie could be made to act like a complete chameleon, producing plots with any kind of aesthetics. And it's fortunately very easy to merge themes in Makie.

**Comment by mkborregaard** [comment]
sorry for belabouring that point if it's obvious

**Comment by asinghvi17** [comment]
That's an interesting point.  By color themes, do you mean:

- colormap, marker color, etc.
- more generally things which affect plots, as opposed to line themes which affect the axis
- dark theme vs light theme is a color theme comparison, BBC vs stock Makie is a line theme comparison

That shouldn't be too hard do do.  I can definitely look into that, especially since color themes seem to share a lot of code!

File: README.md [hunk 1]
```diff
@@ -12,65 +12,66 @@ at https://github.com/cttobin/ggthemr .
...
 ```
 <img src="img/default.png" alt="default" width="500"/>
 
 Using the `:fresh` theme from `GGThemr`
 ```julia
 using MakieThemes
-show_ggthemr(:fresh)
+MakieThemes.demoscatter(:fresh)
 ```
 <img src="img/fresh.png" alt="ggthemr" width="500"/>
```
[the remark this answers is not in the document]
Comment (asinghvi17) [reply README.md:26]: Yeah, I was hoping I could get those hosted in the docs, then just refer to the docs ones from the readme (so the files would live in e.g. `docs/assets/...`).  This way users don't have to download all those PNGs.

[the remark this answers is not in the document]
Comment (asinghvi17) [reply README.md:26]: I did update the AoG plot but nothing beyond that yet.

**Comment by mkborregaard** [comment]
I simply mean the colors. That's the point of the syntax `ggthemr(:colorscheme)`- it signals that I want the fonts, arrows, grid lines etc to look exactly like if this plot was done in R with the https://github.com/Mikata-Project/ggthemr#fresh package, and I want the colors to follow the `:colorscheme`, e.g. `:fresh` for the "fresh" color theme.

So look at the ggthemr package - that one has multiple themes, which are really what I call "color" themes, whereas the ggthemr package itself has a visual identity (compare all the themes on their readme). My idea was to give Makie users complete creative freedom by making it possible to create plots that are visually indistinguishable from what someone would produce with R and ggthemr. You could imagine the same for matplotlib etc.

You can't really emulate that effect with your suggested `theme_bbc()` syntax (which also isn't extensible), and as far as I can see, your BBC plot looks exactly like Makie, just with BBC colors.

The idea is part of a larger vision have  input_format -> syntax -> rendering -> visual expression as completely modular components.

**Comment by asinghvi17** [comment]
Hmm, I see what you mean there.  Here's a possibility:

```
theme_ggthemr(colortheme) = merge(ggthemr(), color_ggthemr(colortheme))
color_ggthemr(colortheme) # color theme (background, axis, etc)
ggthemr() # common 'line' theme
```
and so on for BBC etc.  We can of course change the names if there are some which make more sense.

**Comment by mkborregaard** [comment]
Yes that could work. Or maybe just `theme(style = :ggthemr, color = :fresh)` would be even simpler?

**Comment by asinghvi17** [comment]
I can see the appeal, but my worry is that it would be a little too generic.  We could definitely offer it as an unexported convenience (so it's clear this is `MakieThemes.theme` and not some other `Makie.theme`).

**Comment by asinghvi17** [comment]
I think the docs aren't being pushed up for some reason, but other than that things look good.

**Comment by mkborregaard** [comment]
Trying to run the code in the readme
```
julia> fig
┌ Warning: Can't convert Observable{Symbol} to opengl, for attribute marker
└ @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:427
Error showing value of type Figure:
ERROR: MethodError: no method matching gl_convert(::Symbol)
Closest candidates are:
  gl_convert(::T) where T<:GeometryBasics.Mesh at ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:196
  gl_convert(::T) where T<:ColorTypes.Colorant at ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:194
  gl_convert(::T) where T<:Number at ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:193
  ...
Stacktrace:
  [1] #map#13
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:564 [inlined]
  [2] map
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:562 [inlined]
  [3] const_lift
    @ ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUtils.jl:106 [inlined]
  [4] gl_convert(s::Observable{Symbol})
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:228
  [5] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window, context::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:425
  [6] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:399
  [7] assemble_shader(data::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/glshaders/visualize_interface.jl:108
  [8] draw_lines(screen::Any, position::Observable{Vector{Point{2, Float32}}}, data::Dict)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/glshaders/lines.jl:104
  [9] (::GLMakie.var"#196#197"{GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}})(gl_attributes::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:279
 [10] (::GLMakie.var"#168#171"{GLMakie.var"#196#197"{GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}})()
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:111
 [11] get!(default::GLMakie.var"#168#171"{GLMakie.var"#196#197"{GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}}, h::Dict{UInt64, GLMakie.GLAbstraction.RenderObject}, key::UInt64)
    @ Base ./dict.jl:481
 [12] cached_robj!(robj_func::GLMakie.var"#196#197"{GLMakie.Screen{GLFW.Window}, Scene, Lines{Tuple{Vector{Point{2, Float32}}}}}, screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::Lines{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:84
 [13] draw_atomic
    @ ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:264 [inlined]
 [14] insert!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::Lines{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:126
 [15] insertplots!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:435
 [16] insertplots!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene) (repeats 2 times)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:438
 [17] display_scene!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:374
 [18] GLMakie.Screen(scene::Scene, config::GLMakie.ScreenConfig; visible::Nothing, start_renderloop::Bool)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:391
 [19] GLMakie.Screen(scene::Scene, config::GLMakie.ScreenConfig)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:387
 [20] getscreen(::Module, ::Scene; screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:336
 [21] getscreen
    @ ~/.julia/packages/Makie/gAmAB/src/display.jl:324 [inlined]
 [22] display(figlike::Figure; backend::Module, update::Bool, screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:148
 [23] display(figlike::Figure)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:133
 [24] print_response(errio::IO, response::Any, show_value::Bool, have_color::Bool, specialdisplay::Union{Nothing, AbstractDisplay})
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:0
 [25] (::REPL.var"#45#46"{REPL.LineEditREPL, Pair{Any, Bool}, Bool, Bool})(io::Any)
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:278
 [26] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:521
 [27] print_response(repl::REPL.AbstractREPL, response::Any, show_value::Bool, have_color::Bool)
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:276
 [28] (::REPL.var"#do_respond#66"{Bool, Bool, REPL.var"#77#87"{REPL.LineEditREPL, REPL.REPLHistoryProvider}, REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::Any, ok::Bool)
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:857
 [29] #invokelatest#2
    @ ./essentials.jl:729 [inlined]
 [30] invokelatest
    @ ./essentials.jl:726 [inlined]
 [31] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/LineEdit.jl:2510
 [32] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/REPL/src/REPL.jl:1248
 [33] (::REPL.var"#49#54"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:484
 ```
also
```
julia> MakieThemes.demoscatter(:fresh)
ERROR: MethodError: no method matching merge!(::Symbol, ::Attributes)
Closest candidates are:
  merge!(::Union{Function, Type}, ::Any...) at abstractdict.jl:286
  merge!(::Attributes, ::Attributes...) at ~/.julia/packages/MakieCore/LOKNz/src/attributes.jl:63
Stacktrace:
 [1] set_theme!(new_theme::Symbol; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Makie ~/.julia/packages/Makie/gAmAB/src/theming.jl:153
 [2] set_theme!(new_theme::Symbol)
   @ Makie ~/.julia/packages/Makie/gAmAB/src/theming.jl:151
 [3] with_theme(f::MakieThemes.var"#6#7"{Bool, Bool}, theme::Symbol; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Makie ~/.julia/packages/Makie/gAmAB/src/theming.jl:178
 [4] with_theme(f::Function, theme::Symbol)
   @ Makie ~/.julia/packages/Makie/gAmAB/src/theming.jl:175
 [5] demoscatter(theme::Symbol; legend_vertical::Bool, legend_outside_axis::Bool)
   @ MakieThemes ~/.julia/packages/MakieThemes/EYGqz/src/MakieThemes.jl:73
 [6] demoscatter(theme::Symbol)
   @ MakieThemes ~/.julia/packages/MakieThemes/EYGqz/src/MakieThemes.jl:72
 [7] top-level scope
   @ REPL[19]:1
```

**Comment by asinghvi17** [comment]
ah, second one should be `demoscatter(theme_ggthemr(:fresh))` - will tighten the type constraint on that.

The first one shouldn't be cropping up.  What's your GLMakie/Makie version?

**Comment by mkborregaard** [comment]
```
(@v1.8) pkg> st
Status `~/.julia/environments/v1.8/Project.toml`
  [cbdf2221] AlgebraOfGraphics v0.6.14
  [13f3f980] CairoMakie v0.10.2
  [e9467ef8] GLMakie v0.8.2
  [ee78f7c6] Makie v0.19.2
  [e296ed71] MakieThemes v0.1.0 `https://github.com/JuliaPlots/MakieThemes.jl.git#as/update`
  [5fb14364] OhMyREPL v0.5.13
  [295af30f] Revise v3.5.1
```

**Comment by asinghvi17** [comment]
Huh, `demofigure` seems to work just fine with GLMakie for me (on master).  Have you tried with CairoMakie?

**Comment by asinghvi17** [comment]
OK, things should work pretty well now.  I tested in a closed environment with the same version of GLMakie (v0.8.2) and MakieThemes, and things seemed to run fine, so I'm not sure why the issue you're encountering is triggering.  CairoMakie should work though.

You can download https://github.com/JuliaPlots/MakieThemes.jl/blob/gh-pages/previews/PR16/themes/ggthemr/index.html and open it in a browser to take a look at the docs, since I don't think github pages is activated for this repo.


In terms of the split between color and line themes, here's what I implemented:
```julia
color_ggthemr(:colortheme) # color theme
style_ggthemr()            # line theme
theme_ggthemr(:colortheme) = merge(color_ggthemr(:colortheme), style_ggthemr())
ggthemr(:colortheme) = theme_ggthemr(:colortheme)
```
and the same pipeline for BBC.  I really like this split because you can now do cool stuff like:

```julia
MakieThemes.demofigure(merge(style_bbc(), color_ggthemr(:carrot)))
```
Screen Shot 2023-02-09 at 7 23 07 AM

**Comment by mkborregaard** [comment]
So does the code currently in the Readme.md file work for you?

**Comment by asinghvi17** [comment]
Yeah, all of it works fine for me.  CI is also running, so I don't think there's an incompatibility with Makie, and all the themes are run and shown in the docs (link is https://makieorg.github.io/MakieThemes.jl/dev).

**Pull request merged**